src/geo/ui/widgets/time-series/torque-histogram-view.js

```diff
@@ -0,0 +1,124 @@
+var $ = require('jquery');
+var _ = require('underscore');
+var Model = require('cdb/core/model');
+var View = require('cdb/core/view');
+var HistogramChartView = require('../histogram/chart');
+var TorqueTimeMarkerview = require('./torque-time-marker-view');
+
+/**
+ * Torque time-series histogram view.
+ * Extends the common histogram chart view with time-control
+ */
+module.exports = View.extend({
+
+  // TODO could be calculated from element styles instead of duplicated numbers here?
+  defaults: {
+    width: 400,
+    margins: {
+      top: 0,
+      right: 24,
+      bottom: 0,
+      left: 24
+    }
+  },
+
+  initialize: function() {
+    if (!this.options.torqueLayerModel) throw new Error('torqeLayerModel is required');
+    if (!this.options.filter) throw new Error('filter is required');
+
+    this._filter = this.options._filter;
+
+    _.bindAll(this, '_onWindowResize');
+    $(window).bind('resize', this._onWindowResize);
+
+    this._viewModel = new Model({
+      width: this.defaults.width,
+      histogramChartMargins: {
+        top: 4,
+        right: 4,
+        bottom: 20,
+        left: 4
+      },
+      margins: this.defaults.margins,
+      histogramChartHeight:
+        48 + // inline bars height
+        20 + // bottom labels
+        4 // margins
+    });
+    this._viewModel.bind('change:width', this._onChangeWidth, this);
+    this.add_related_model(this._viewModel);
+
+    // TODO w/o this the timemarker ends up behind the removed-and-added elements in the histogram chart,
+    // can we avoid that?
+    this.model.bind('change:data', this.render, this);
+    this._viewModel.bind('change:width', this.render, this);
```

Comment:
Do you need to re-render the widget every time the data is changed? In the case of the plain histogram I only render it once and then use the `refresh` and `reset` methods when I want to animate the height of the bars to the new values or regenerate the whole chart respectively.